evm/src/cpu/kernel/assembler.rs

```diff
@@ -41,33 +55,31 @@ pub(crate) fn assemble(files: Vec<File>) -> Kernel {
     }
 }
 
-fn find_macros(files: &[File]) -> HashMap<String, Vec<Item>> {
+fn find_macros(files: &[File]) -> HashMap<String, Macro> {
     let mut macros = HashMap::new();
     for file in files {
         for item in &file.body {
-            if let Item::MacroDef(name, items) = item {
-                macros.insert(name.clone(), items.clone());
+            if let Item::MacroDef(name, params, items) = item {
+                let _macro = Macro {
```

Comment:
nit: Why the underscore?

Reply:
`macro` seems to be reserved in Rust (i.e. not a keyword yet, but reserved for future use)

Reply:
Ah I see, I've usually seen people put the underscore after the name in these cases. You can also escape keywords using `r#macro`, although it's pretty ugly. Anyways, definitely not important.